Add component generator (#450)
primer/view_components: CONTRIBUTING.md

Please note that this project is released with a [Contributor Code of Conduct](CODE_OF_CONDUCT.md). By participating in this project you agree to abide by its terms.

## Creating a component

You can use a generator to create a component and add it to all the necessary files.

```sh
bundle exec thor component_generator my_component_name
```

> If your component requires a JS package, you can pass the `js` flag to the generator.

Reviewer: ```suggestion
To declare a dependency on an `npm` package, pass `js` to the generator:
```
